src/Sarif/Writers/PrereleaseCompatibilityTransformer.cs

```diff
@@ -188,9 +188,88 @@ private static bool ApplyBeta3Changes(JObject sarifLog)
             // https://github.com/oasis-tcs/sarif-spec/issues/399
             modifiedLog |= ConvertInvocationToolExecutionSuccessfulToExecutionSuccessful(sarifLog);
 
+            // https://github.com/oasis-tcs/sarif-spec/issues/401
+            modifiedLog |= RenameBaseAddressAndOffsetInAddressObject(sarifLog);
+
             return modifiedLog;
         }
 
+        private static bool RenameBaseAddressAndOffsetInAddressObject(JObject sarifLog)
+        {
+            string[] addressPathsToUpdate =
+            {
+                "inlineExternalProperties[].addresses[]",
```

Comment:
>"inlineExternalProperties[].addresses[]", [](start = 16, length = 41)

we have all these address usages! the reason why i didnt use the simple rename function is because we have another object ("artifact") with a property named "offset". If we know for a fact that some of these objects are not being used, i can comment them out to avioud unneeded scans (e.g. all the inlineExternalProperties?)